Microsoft.Toolkit.Win32.UI.Controls/Interop/WinRT/WebViewControlProcess.cs

```diff
@@ -0,0 +1,144 @@
+// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Security;
+using System.Threading.Tasks;
+
+using Windows.Foundation;
+using Windows.Web.UI.Interop;
+
+namespace Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT
+{
+
+    /// <summary>
+    /// A proxy for <see cref="Windows.Web.UI.Interop.WebViewControlProcess"/>.
+    /// </summary>
+    public class WebViewControlProcess
+    {
+        [SecurityCritical]
+        private readonly Windows.Web.UI.Interop.WebViewControlProcess _process;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebViewControlProcess"/> class.
+        /// </summary>
+        public WebViewControlProcess()
+            : this(new WebViewControlProcessOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebViewControlProcess"/> class with the specified <paramref name="processOptions"/>.
+        /// </summary>
+        /// <param name="processOptions">The process options.</param>
+        public WebViewControlProcess(WebViewControlProcessOptions processOptions)
+            : this(new Windows.Web.UI.Interop.WebViewControlProcess(processOptions.ToWinRtWebViewControlProcessOptions()))
+        {
+        }
+
+        private WebViewControlProcess(Windows.Web.UI.Interop.WebViewControlProcess process)
+        {
+            _process = process ?? throw new ArgumentNullException(nameof(process));
+            SubscribeEvents();
+        }
+
+        /// <summary>
+        /// Occurs when the underlying WWAHost process exits.
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1009:DeclareEventHandlersCorrectly")]
+        public event EventHandler<object> ProcessExited = (sender, args) => { };
```

Comment:
@normesta @mcleanbyron This is strange since there are two processes that actually get created: Win32WebViewHost.exe and WWAHost.exe. the PID we get back and the notifications we get are for WWAHost.exe, but it is more often the case that we have a problem with Win32WebViewHost.exe and need to respond to that. There are cases in RS4 where WWAHost.exe may be orphaned because of a crash (see our bug list) and we may not be notified because WWAHost.exe is fine, or we are delayed in receiving the notification because WWAHost has not yet caught on to the crash.